### DISCUSSION TITLE
MySQL Connection Error

### DISCUSSION THREAD

**Discussion by 641i130** [Pryaxis/TShock#2200] [Q&A] [answered]
``2021-01-08_22-51-38.log  2021-01-08_23-01-34.log  2021-01-08_23-13-16.log  config.json	Dimensions.json  motd.txt  old	rules.txt  setup.lock  sscconfig.json  tshock.pid  whitelist.txt
root@mainhost:/home/carrot/T/Terraria/main/tshock# cat 2021-01-08_22-51-38.log
2021-01-08 22:51:38 - TShock: INFO: TShock was improperly shut down. Please use the exit command in the future to prevent this.
2021-01-08 22:51:38 - <>c: INFO: Port overridden by startup argument. Set to 7780
2021-01-08 22:51:39 - TShock: ERROR: Fatal Startup Exception
2021-01-08 22:51:39 - TShock: ERROR: System.Exception: Fatal TShock initialization exception: failed to connect to MySQL database. See inner exception for details. ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> Mono.Btls.MonoBtlsException: Ssl error:1000042e:SSL routines:OPENSSL_internal:TLSV1_ALERT_PROTOCOL_VERSION
  at /build/mono-6.12.0.107/external/boringssl/ssl/tls_record.c:462
  at Mono.Btls.MonoBtlsContext.ProcessHandshake () [0x00048] in <6bc04dcac0a443ee834a449c98b8ed9d>:0 
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake (Mono.Net.Security.AsyncOperationStatus status, System.Boolean renegotiate) [0x000da] in <6bc04dcac0a443ee834a449c98b8ed9d>:0 
  at (wrapper remoting-invoke-with-check) Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake(Mono.Net.Security.AsyncOperationStatus,bool)
  at Mono.Net.Security.AsyncHandshakeRequest.Run (Mono.Net.Security.AsyncOperationStatus status) [0x00006] in <6bc04dcac0a443ee834a449c98b8ed9d>:0 
  at Mono.Net.Security.AsyncProtocolRequest.ProcessOperation (System.Threading.CancellationToken cancellationToken) [0x000fc] in <6bc04dcac0a443ee834a449c98b8ed9d>:0 
   --- End of inner exception stack trace ---
  at Mono.Net.Security.MobileAuthenticatedStream.AuthenticateAsClient (System.String targetHost, System.Security.Cryptography.X509Certificates.X509CertificateCollection clientCertificates, System.Security.Authentication.SslProtocols enabledSslProtocols, System.Boolean checkCertificateRevocation) [0x0004b] in <6bc04dcac0a443ee834a449c98b8ed9d>:0 
  at (wrapper remoting-invoke-with-check) Mono.Net.Security.MobileAuthenticatedStream.AuthenticateAsClient(string,System.Security.Cryptography.X509Certificates.X509CertificateCollection,System.Security.Authentication.SslProtocols,bool)
  at System.Net.Security.SslStream.AuthenticateAsClient (System.String targetHost, System.Security.Cryptography.X509Certificates.X509CertificateCollection clientCertificates, System.Security.Authentication.SslProtocols enabledSslProtocols, System.Boolean checkCertificateRevocation) [0x00006] in <6bc04dcac0a443ee834a449c98b8ed9d>:0 
  at MySql.Data.MySqlClient.NativeDriver.StartSSL () [0x0002f] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at MySql.Data.MySqlClient.NativeDriver.Open () [0x002c5] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at MySql.Data.MySqlClient.Driver.Open () [0x0000b] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at MySql.Data.MySqlClient.Driver.Create (MySql.Data.MySqlClient.MySqlConnectionStringBuilder settings) [0x0004d] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnection () [0x00000] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at MySql.Data.MySqlClient.MySqlPool.GetPooledConnection () [0x0008a] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at MySql.Data.MySqlClient.MySqlPool.TryToGetDriver () [0x00042] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at MySql.Data.MySqlClient.MySqlPool.GetConnection () [0x0001c] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at MySql.Data.MySqlClient.MySqlConnection.Open () [0x0016c] in <15029c3817b245ffb50e9433f5ade25d>:0 
  at TShockAPI.DB.DbExt.QueryReader (System.Data.IDbConnection olddb, System.String query, System.Object[] args) [0x00007] in <2ac9e320a8aa42aeba476df6376f5727>:0 
   --- End of inner exception stack trace ---
  at TShockAPI.DB.DbExt.QueryReader (System.Data.IDbConnection olddb, System.String query, System.Object[] args) [0x0006b] in <2ac9e320a8aa42aeba476df6376f5727>:0 
  at TShockAPI.DB.SqlTableCreator.GetColumns (TShockAPI.DB.SqlTable table) [0x00093] in <2ac9e320a8aa42aeba476df6376f5727>:0 
  at TShockAPI.DB.SqlTableCreator.EnsureTableStructure (TShockAPI.DB.SqlTable table) [0x0000d] in <2ac9e320a8aa42aeba476df6376f5727>:0 
  at TShockAPI.DB.BanManager..ctor (System.Data.IDbConnection db) [0x000dc] in <2ac9e320a8aa42aeba476df6376f5727>:0 
  at TShockAPI.TShock.Initialize () [0x003b8] in <2ac9e320a8aa42aeba476df6376f5727>:0 ``

**MySQL Version:** `mysql  Ver 8.0.22 for Linux on x86_64 (MySQL Community Server - GPL)`

**VPS Debian Version:** `Debian GNU/Linux 10`

**Linux Kernel:** `4.19.0-13-amd64`

Basically, I set up a MySQL server and migrated over my sqlite3 file using this [python program](https://pypi.org/project/sqlite3-to-mysql/). Now I'm at the point of trying to connect but no beans. Any ideas? Should I downgrade my sql version? What version would be appropiate.

**Comment by 641i130** [comment, marked as answer]
After further research, an option to disable SSL would solve this issue. I'm not sure how I would do that but I'm assuming its done in the source code since theres no option in config.js
